src/TestFramework/Coverage/JUnit/JUnitTestFileDataProvider.php

```diff
@@ -65,23 +65,12 @@ public function getTestFileInfo(string $fullyQualifiedClassName): TestFileTimeDa
     {
         $xPath = $this->getXPath();
 
-        $nodes = $xPath->query(
-            sprintf('//testsuite[@name="%s"]', $fullyQualifiedClassName)
-        );
+        foreach (self::testCaseMapGenerator($fullyQualifiedClassName) as $queryString => $placeholder) {
+            $nodes = $xPath->query(sprintf($queryString, $placeholder));
 
-        if ($nodes->length === 0) {
-            // Try another format where the class name is inside `class` attribute of `testcase` tag
-            $nodes = $xPath->query(
-                sprintf('//testcase[@class="%s"]', $fullyQualifiedClassName)
-            );
-        }
-
-        if ($nodes->length === 0) {
-            $feature = preg_replace('/^(.*):+.*$/', '$1.feature', $fullyQualifiedClassName);
-            // try another format where the class name is inside `file` attribute of `testcase` tag
-            $nodes = $xPath->query(
-                sprintf('//testcase[contains(@file, "%s")]', $feature)
-            );
+            if ($nodes->length !== 0) {
+                break;
+            }
         }
 
         if ($nodes->length === 0) {
```

Comment:
Scrutinizer:

>The variable $nodes seems to be defined by a foreach iteration on line 68. Are you sure the iterator is never empty, otherwise this variable is not defined?

Reply:
We're 100% sure about that.